google/cloud/log_test.cc

```diff
@@ -292,6 +292,11 @@ TEST(LogSinkTest, DisabledLogsMakeNoCalls) {
   EXPECT_EQ(0, counter);
 }
 
+TEST(LogFatalTest, NoReturn) {
+  LogSink::EnableStdClog();
```

Comment:
Is this needed because the messages are not logged otherwise?  There is something odd about crashing the application without logging.  Maybe `FATAL` should always log to `std::clog`?  Or maybe it should log to `std::clog` if there are no other backends configured?

Reply:
I concur that having `FATAL` logs (or indeed, perhaps anything about `WARNING`) always at least end up on `std::clog` might be a feature.  But this PR is not changing that behavior.  Even before a `GCP_LOG(FATAL) << "foo"; std::exit(1)` could silently exit if there was no enabled log sink.  So, I think we should consider that separately.

Reply:
Yeah, I did not mean you should fix that in this PR.  Separate change for it SGTM too.